### PR TITLE
Be ready for PHP 7.0!

### PR DESCRIPTION
**Pull request by renatomefi**
Since PHP 7.0 is almost here and this lib is really new, why not be prepared to PHP 7?

I'm updating Travis to test 5.4 -> 7.0, 5.3 is excluded since the tests' dataprovider are using the new array notation, if we want to use PHP 5.3 the old php notation should be used!

In order to be compatible, was needed to change some classes names, like: Bool, String, Int, Float. Why?
https://wiki.php.net/rfc/reserve_more_types_in_php_7
http://php.net/manual/en/reserved.other-reserved-words.php

My proposal is to use the prefix Cast for those classes, minus Bool, which is now Boolean!

### CODE REVIEW

**Comment by rick-nu** [comment]
You're absolutely right, we should be prepared for PHP 7! There are some dependencies from you other PR in this PR though, I would like to add the Travis checks to the repo. If you can strip the dependency and we can make sure that Travis still builds (as I see no Travis builds on the latest commits), I would like to merge this :)

**Comment by rick-nu** [comment]
> In order to be compatible, was needed to change some classes names, like: Bool, String, Int, Float. Why?
> https://wiki.php.net/rfc/reserve_more_types_in_php_7
> http://php.net/manual/en/reserved.other-reserved-words.php

Very nice tweaks by the way :)

**Comment by renatomefi** [comment]
Hello @RickvdStaaij, sorry for that, I did the checkout from the wrong branch.
Now it's rebased and updated!

**Comment by rick-nu** [comment]
Perfect! Thanks a lot! Also, I'm looking forward to see you in the Netherlands and you joining Enrise :D

**Comment by renatomefi** [comment]
Thank you @RickvdStaaij, I'm also pretty anxious to join Enrise, you all guys seems to be very welcoming people! :)

File: src/FilterResource.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function append($append)
      */
     public function bool()
     {
-        return $this->addRule(new FilterRule\Bool);
+        return $this->addRule(new FilterRule\Boolean);
```
Comment (NickBelhomme) [inline src/FilterResource.php:55]: I know technically this resolves the reserved words incompatibility, but why not use the same naming scheme as all the others: CastBool?

Comment (rick-nu) [inline src/FilterResource.php:55]: I agree, missed that one, will fix in a future feature.

**Pull request merged**